lib/transbank/sdk/onepay/models/shopping_cart.rb

```diff
@@ -1,68 +1,80 @@
-require 'transbank/sdk/onepay/utils/jsonify'
-
 module Transbank
   module Onepay
-    # TODO: YARD this class
+    # Represents a Shopping Cart, which contains [Item]s that the user wants to buy
     class ShoppingCart
-      include Utils::JSONifier
-      # Represents a Shopping Cart, which contains [Item]s that the user wants to buy
-
-
+      include Utils::JSONUtils
       # @return [Integer] The amount in CLP of the [Item]s included in the [ShoppingCart]
       attr_reader :total
 
-      def initialize
+      # @param items [Array, nil] an array of Hashes that can be converted to [Item]
+      # if nil, an empty shopping cart is created
+      def initialize(items = [])
         # The amount in CLP of the [Item]s included in the [ShoppingCart]
         @total = 0
         # An [Array<Item>] with the [ShoppingCart] contents
         @items = []
+        return if items.nil? || items.empty?
+
+        items.each do |it|
+          it = transform_hash_keys it
+          item = Item.new it
+          self << item
+        end
       end
 
       # @param item [Item] an instance of [Item]
       # @return [boolean] return true if item is successfully added
       def add(item)
-        new_total = calculate_total(item)
+        new_total = total + item.total
+        if new_total < 0
+          raise Errors::ShoppingCartError, "New total amount cannot be less than zero."
+        end
         @items << item
         @total = new_total
         true
       end
 
-      # Call #add(item) on self, see docs for #add
+      # Alias for #add
       def << item
         add item
       end
 
+      # Remove an [Item] from self
+      # @return [boolean] return true if the item is successfully removed
       def remove(item)
-        new_total = calculate_total(item)
+        new_total = total - item.total
+        if new_total < 0
+          raise Errors::ShoppingCartError, "New total amount cannot be less than zero."
+        end
         first_instance_of_item = @items.index(item)
+        if first_instance_of_item.nil?
+          raise Errors::ShoppingCartError, 'Item not found'
+        end
         @items.delete_at first_instance_of_item
         @total = new_total
         true
       end
 
+      # Clear the cart, setting @total to 0 and @items to []
+      # @return [boolean] true
       def remove_all
         @total = 0
         @items = []
+        true
```

Comment:
Why you return something if it's always the same value?

Reply:
Right about when we were to deliver the original SDKs we noticed that if we returned a reference to the items, you could modify the items and make the ShoppingCart's total not be equal to the sum of the items' totals.

A solution for this is having "total" be a method that calculates the total on the spot, instead of a property

Reply:
By doing total as a method that calcs the total on the spot by iterating over the items we also remove the need for the == method, since we can then use direct references to the items

Reply:
Yup, sounds like keeping state for `total` is a bad tradeoff. It's not like it's a too expensive operation nor like people will be calling it too often.